godtools/App/Features/ChooseYourOwnAdventure/ChooseYourOwnAdventureViewModel.swift

```diff
@@ -28,16 +29,14 @@ class ChooseYourOwnAdventureViewModel: MobileContentPagesViewModel, ChooseYourOw
         
         let navBarColor: UIColor
         let navBarControlColor: UIColor
+
+        let manifest: Manifest? = renderer.pageRenderers.first?.manifest
         
-        if let manifestAttributes = renderers.first?.parser.manifest.attributes,
-            let barColor = manifestAttributes.navbarColor?.uiColor,
-            let controlColor = manifestAttributes.navbarControlColor?.uiColor {
-            
-            navBarColor = barColor
-            navBarControlColor = controlColor
+        if let manifest = manifest {
+            navBarColor = manifest.navBarColor
+            navBarControlColor = manifest.navBarControlColor
         }
         else {
-            
             navBarColor = ColorPalette.gtBlue.color
             navBarControlColor = .white
```

Comment:
hmm, we should talk about how you are able use the default colors that are present in the parser.

within Kotlin the `navBarColor` and `navBarControlColor` properties use a nullable receiver, so if my manifest is null they will resolve to the default colors. But I don't know if that concept is translated to Swift or if we will need to expose those defaults in another manner.

```kotlin
val manifest: Manifest? = null
manifest.navBarColor // this would return the default color because manifest is null
```

Reply:
Hey @frett I haven't heard of the nullable receiver concept before.  I actually need to update that because I changed the behavior for opening a tool.  If the manifest is null an error is presented instead of opening to a tool.

Reply:
alright, let me know if there are any other places that you could benefit from having the defaults values from the shared parser

Reply:
the Kotlin nullable receiver concept may not be something that translates into Swift, in which case we will probably want to figure out an alternate way to expose those defaults